hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/security/OMRenewDelegationTokenRequest.java

```diff
@@ -142,13 +151,18 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
     } catch (IOException ex) {
       LOG.error("Error in Updating Renew DelegationToken {}",
           ozoneTokenIdentifierToken, ex);
+      exception = ex;
       omClientResponse = new OMRenewDelegationTokenResponse(null, -1L,
           createErrorOMResponse(omResponse, ex));
     } finally {
       addResponseToDoubleBuffer(transactionLogIndex, omClientResponse,
           ozoneManagerDoubleBufferHelper);
     }
 
+    auditLog(auditLogger,
+        buildAuditMessage(OMAction.GET_DELEGATION_TOKEN, auditMap, exception,
```

Comment:
I have tried to address the issue of preExecute failure not being logged in audit logs in my latest patch. Please let me know if it's the correct way to handle it.

Reply:
Thanks @JyotinderSingh for updating the patch, changes look good.  Regarding failure logging, I think we can improve it a bit.

I have executed the following test:

```
kinit -kt /etc/security/keytabs/testuser.keytab testuser/scm@EXAMPLE.COM
ozone sh token get
kinit -kt /etc/security/keytabs/testuser2.keytab testuser2/scm@EXAMPLE.COM
ozone sh token renew
```

Result in audit log:

```
ERROR | OMAudit | user= | ip= | op=RENEW_DELEGATION_TOKEN {username=, clientID=client-17A77FF07B03} | ret=FAILURE | TOKEN_ERROR_OTHER org.apache.hadoop.ozone.om.exceptions.OMException: Delegation token renewal failed for dt: Kind: OzoneToken, Service: 192.168.240.2:9862, Ident: 08 01 1a 17 74 65 73 74 75 73 65 72 2f 6f 6d 40 45 58 41 4d 50 4c 45 2e 43 4f 4d 22 08 74 65 73 74 75 73 65 72 2a 00 30 90 c7 a7 9f f9 2f 38 90 cf d9 bf fb 2f 40 02 48 01 5a 0f 34 30 37 31 31 34 38 33 38 34 35 30 30 35 37 7a 12 6f 6d 53 65 72 76 69 63 65 49 64 44 65 66 61 75 6c 74
	at org.apache.hadoop.ozone.om.OzoneManager.renewDelegationToken(OzoneManager.java:2164)
	at org.apache.hadoop.ozone.om.request.security.OMRenewDelegationTokenRequest.preExecute(OMRenewDelegationTokenRequest.java:77)
```

User information (username, IP) is missing.  It can be fixed by getting and using the updated request by calling `super.preExecute`.  Example from `OMRenewDelegationTokenRequest`:

```
  @Override
  public OMRequest preExecute(OzoneManager ozoneManager) throws IOException {
    OMRequest request = super.preExecute(ozoneManager);

    RenewDelegationTokenRequestProto renewDelegationTokenRequest =
        request.getRenewDelegationTokenRequest();

    AuditLogger auditLogger = ozoneManager.getAuditLogger();
    Map<String, String> auditMap = null;

    long renewTime;
    try {
      Token<OzoneTokenIdentifier> token = OMPBHelper.convertToDelegationToken(
          renewDelegationTokenRequest.getToken());
      auditMap = buildTokenAuditMap(token);

      // Call OM to renew token
      renewTime = ozoneManager.renewDelegationToken(token);
    } catch (IOException ioe) {
      if (auditMap == null) {
        auditMap = new LinkedHashMap<>();
      }
      auditLog(auditLogger,
          buildAuditMessage(OMAction.RENEW_DELEGATION_TOKEN, auditMap, ioe,
              request.getUserInfo()));
      throw ioe;
    }
...
```

With this, audit log contains similar information for failure as it does for success:

```
ERROR | OMAudit | user=testuser2/scm@EXAMPLE.COM | ip=172.19.0.2 | op=RENEW_DELEGATION_TOKEN {kind=OzoneToken, service=172.19.0.10:9862} | ret=FAILURE | TOKEN_ERROR_OTHER ...
```

Similar change can be applied to the other requests.

Reply:
Thank you for the suggestion @adoroszlai, I have updated the patch with these changes.